src/mgmt_conn.c

```diff
@@ -1042,16 +1042,40 @@ uzfs_zvol_execute_async_command(void *arg)
 		break;
 	case ZVOL_OPCODE_RESIZE:
 		volsize = *(uint64_t *)async_task->payload;
-		rc = zvol_check_volsize(volsize,
-		    zinfo->main_zv->zv_volblocksize);
-		if (rc == 0) {
-			rc = zvol_update_volsize(volsize,
-			    zinfo->main_zv->zv_objset);
-			if (rc == 0)
-				zvol_size_changed(zinfo->main_zv, volsize);
+		// Take rebuild_mtx lock since we are checking the status
+		mutex_enter(&zinfo->main_zv->rebuild_mtx);
```

Comment:
I don't think we need lock here

Reply:
During state transmission time, there might be chances for volume resize request. So it is required

Reply:
sounds good with the lock.. lets have a log for this before the start of lock and at the end for success case as well

Reply:
We are already having a log in process_message for resize triggered case.

Reply:
Added successful log message